apollo-federation/src/sources/connect/validation/selection.rs

```diff
@@ -56,6 +57,33 @@ pub(super) fn validate_selection(
     .err()
 }
 
+pub(super) fn validate_body_selection(
+    connect_directive: &Node<Directive>,
+    parent_type: &Node<ObjectType>,
+    field: &Component<FieldDefinition>,
+    schema: &Schema,
+    selection_node: &Node<Value>,
+) -> Result<(), Message> {
+    let coordinate =
+        connect_directive_http_body_coordinate(&connect_directive.name, parent_type, &field.name);
+
+    let selection_str = require_value_is_str(&selection_node, &coordinate, &schema.sources)?;
+
+    if selection_str.is_empty() {
```

Comment:
Is something before this removing any whitespace so something like `selection: "   "` will be considered empty?

Reply:
Oh good call, maybe here and in body we should parse JSONSelection and then check if it's empty? Probably means putting in helper method in selection which returns true for _any_ expression which is semantically empty? Since a selection containing only a comment is still empty.

Reply:
great idea, i added a is_empty() method on JSONSelection instead